xio-core/src/main/java/com/xjeffrose/xio/http/ClientConnectionManager.java

```diff
@@ -0,0 +1,82 @@
+package com.xjeffrose.xio.http;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandler;
+import java.util.function.Supplier;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ClientConnectionManager {
+  private ClientChannelInitializer channelInitializer;
+  private ChannelFutureListener releaseListener;
+  private ChannelFutureListener connectionListener;
+  private ChannelFuture currentChannelFuture;
+  private ClientState state;
+  private boolean reusable = true;
+  private ClientConnectionState connectionState = ClientConnectionState.NOT_CONNECTED;
+
+  Channel currentChannel() {
+    if (currentChannelFuture != null) {
+      return currentChannelFuture.channel();
+    }
+    return null;
+  }
+
+  public ClientConnectionManager(ClientState state, ClientChannelInitializer channelInitializer) {
+    this.state = state;
+    this.channelInitializer = channelInitializer;
+
+    releaseListener =
+        f -> {
+          log.debug("Channel closed");
+          connectionState = ClientConnectionState.CLOSED_CONNECTION;
+          this.currentChannelFuture = null;
+          this.reusable = false;
+        };
+    connectionListener =
+        f -> {
```

Comment:
Do you mind changing the variable name to be more descriptive than a single letter?

Reply:
Same question a few lines above as well